### ISSUE TITLE
Add Prologue/Epilogue Options to Script

### ISSUE THREAD

**Issue by MorganRodgers** [OSC/ood_core#140]
All currently supported schedulers offer the ability to run tasks just before and just after a job:

- [GridEngine](http://www.softpanorama.org/HPC/Grid_engine/prolog_and_epilog_scripts.shtml) (unofficial docs)
- [LSF](http://www.redbooks.ibm.com/redpapers/pdfs/redp5048.pdf) (reference in PDF)
- [PBSPro](https://www.pbsworks.com/pdfs/PBSUserGuide14.2.pdf) (reference in PDF)
- [Slurm](https://slurm.schedmd.com/prolog_epilog.html)
- [Torque](http://docs.adaptivecomputing.com/torque/6-1-2/adminGuide/torque.htm#topics/torque/13-appendices/gScripts/momScripts.htm%3FTocPath%3DAppendices%7CAppendix%2520G%253A%2520%25C2%25A0%2520Prologue%2520and%2520Epilogue%2520Scripts%7C_____1)

The ability to run clean up tasks in a separate task would be useful for BatchConnect applications in cases where `trap` does not work.

**Comment by johrstrom** [comment]
Good idea, but it's not clear to me if this is a user privilege or if (for example in the Slurm) it requires the admin to supply them.

**Comment by moffatsadeghi** [comment]
GitHub Review Meeting Summary from May: 
A request (originally opened by Morgan, a former OSC staff member) to add true prologue and epilogue script options that run before and after the actual job, distinct from the existing before/after scripts that run inside the job.
Details
    bsingleton: Clarified the distinction: current before/after scripts run inside the job; true prologue/epilogue would run outside, which is what users often expect.
    bsingleton: Suggested epilogue could help with collecting job exit status, a recurring community request.
    johrstrom: Noted that job exit status collection is already addressed via the historic info API (sacct vs squeue for Slurm); pointed out that prologue/epilogue in Slurm may be an administrator-level privilege, not user-level.
    johrstrom: Confirmed he regularly references epilogues/prologues on Discourse as a solution for users.
Conclusion
    Issue kept open, not scheduled.
    "Needs investigation" label suggested to determine whether prologue/epilogue is user-level or admin-level across schedulers.
